src/View/Helper/FormHelper.php

```diff
@@ -286,19 +292,35 @@ public function __construct(View $View, array $config = [])
      * @param \Cake\View\Widget\WidgetRegistry|null $instance The registry instance to set.
      * @param array $widgets An array of widgets
      * @return \Cake\View\Widget\WidgetRegistry
+     * @deprecated 3.6.0 Use FormHelper::widgetLocator() instead.
      */
     public function widgetRegistry(WidgetRegistry $instance = null, $widgets = [])
+    {
+        deprecationWarning('widgetRegistry is deprecated, use widgetLocator instead.');
+
+        return $this->widgetLocator($instance, $widgets);
+    }
+
+    /**
+     * Set the widget registry the helper will use.
```

Comment:
also this method returns an object of  `Cake\View\Widget\WidgetLocator` 
we can not split this method to setter/getter IMO
because we have now `addWidget()`

Reply:
Do you mean that both `widgetRegistry` and `widgetLocator` should return `Cake\View\Widget\WidgetLocator`?

I think it shuoldn't be a problem to have a getter/setter while keeping `addWidget`, but I'll wait for comments

Reply:
>Do you mean that both widgetRegistry and widgetLocator should return Cake\View\Widget\WidgetLocator?

no , i mean this method is not just setter 

https://book.cakephp.org/3.0/en/appendices/3-5-migration-guide.html#deprecated-combined-get-set-methods
https://book.cakephp.org/3.0/en/appendices/3-4-migration-guide.html#deprecated-combined-get-set-methods
we want to split methods into setter/getter methods where possible , like methods in above links 
but for this method  I'm not sure  @markstory

Reply:
Splitting this method into `setWidgetLocator()` and `getWidgetLocator()` would match the pattern we're using elsewhere better. Having the separate methods will allow us to add return types more easily in 4.x.